web/src/site-admin/SiteAdminExportConfigPage.tsx

```diff
@@ -0,0 +1,146 @@
+import siteSchemaJSON from '../../../schema/site.schema.json'
+import { PageTitle } from '../components/PageTitle'
+import { ExternalServiceKind } from '../../../shared/src/graphql/schema'
+
+/**
+ * Minimal shape of a JSON Schema. These values are treated as opaque, so more specific types are
+ * not needed.
+ */
+interface JSONSchema {
+    $id: string
+    definitions?: Record<string, { type: string }>
+}
+
+const externalServices: [ExternalServiceKind, JSONSchema][] = [
+    [ExternalServiceKind.AWSCODECOMMIT, awsCodeCommitJSON],
+    [ExternalServiceKind.BITBUCKETCLOUD, bitbucketCloudSchemaJSON],
+    [ExternalServiceKind.BITBUCKETSERVER, bitbucketServerSchemaJSON],
+    [ExternalServiceKind.GITHUB, githubSchemaJSON],
+    [ExternalServiceKind.GITLAB, gitlabSchemaJSON],
+    [ExternalServiceKind.GITOLITE, gitoliteSchemaJSON],
+    [ExternalServiceKind.OTHER, otherExternalServiceSchemaJSON],
+    [ExternalServiceKind.PHABRICATOR, phabricatorSchemaJSON],
+]
+
+const allConfigSchema = {
+    $id: 'all.schema.json#',
+    allowComments: true,
+    additionalProperties: false,
+    properties: {
+        site: siteSchemaJSON,
+        externalServices: {
+            type: 'object',
+            properties: externalServices.reduce<
+                Partial<{ [k in ExternalServiceKind]: { type: string; items: JSONSchema } }>
+            >((properties, [kind, schema]) => {
+                properties[kind] = {
+                    type: 'array',
+                    items: schema,
+                }
+                return properties
+            }, {}),
+        },
+        settings: {
+            type: 'object',
+            properties: {
+                subjects: {
+                    type: 'array',
+                    items: {
+                        type: 'object',
+                        properties: {
+                            __typename: {
+                                type: 'string',
+                            },
+                            settingsURL: {
+                                type: ['string', 'null'],
+                            },
+                            contents: {
+                                ...settingsSchemaJSON,
+                                type: ['object', 'null'],
+                            },
+                        },
+                    },
+                },
+                final: settingsSchemaJSON,
+            },
+        },
+    },
+    definitions: externalServices
+        .map(([, schema]) => schema.definitions)
+        .concat([siteSchemaJSON.definitions, settingsSchemaJSON.definitions])
+        .reduce(
+            (definitions, theseDefinitions) =>
+                theseDefinitions
+                    ? {
+                          ...definitions,
+                          ...theseDefinitions,
+                      }
+                    : definitions,
+            {}
+        ),
+}
+
+interface Props extends RouteComponentProps<any> {
```

Comment:
`RouteComponentProps` has a default value for `Params`, `{}`. Omit the type parameter to default to that value, rather than using `any`.

```suggestion
interface Props extends RouteComponentProps {
```